apps/files_external/lib/Migration/Version1015Date20211104103506.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Files_External\Migration;
+
+use Closure;
+use OC\Files\Cache\Storage;
+use OCP\DB\Exception;
+use OCP\DB\IResult;
+use OCP\DB\ISchemaWrapper;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+use Psr\Log\LoggerInterface;
+
+class Version1015Date20211104103506 extends SimpleMigrationStep {
+
+	/** @var IDBConnection */
+	private $connection;
+	/** @var LoggerInterface */
+	private $logger;
+
+	public function __construct(IDBConnection $connection, LoggerInterface $logger) {
+		$this->connection = $connection;
+		$this->logger = $logger;
+	}
+
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		$qb = $this->connection->getQueryBuilder();
+		$qb->update('storages')
+			->set('id', $qb->createParameter('newId'))
+			->where($qb->expr()->eq('id', $qb->createParameter('oldId')));
+
+		$mounts = $this->getS3Mounts();
+		if (!$mounts instanceof IResult) {
+			throw new \Exception('Could not fetch existing mounts for migration');
+		}
+
+		while ($mount = $mounts->fetch()) {
+			$config = $this->getStorageConfig((int)$mount['mount_id']);
+			$hostname = $config['hostname'];
+			$bucket = $config['bucket'];
+			$key = $config['key'];
+			$oldId = Storage::adjustStorageId('amazon::' . $bucket);
+			$newId = Storage::adjustStorageId('amazon::external::' . md5($hostname . ':' . $bucket . ':' . $key));
+			try {
+				$qb->setParameter('oldId', $oldId);
+				$qb->setParameter('newId', $newId);
+				$qb->executeStatement();
```

Comment:
If you want to backport this, you probably should use this instead:

```suggestion
				$qb->execute();
```